src/components/editgrid/EditGrid.js

```diff
@@ -402,7 +423,7 @@ export default class EditGridComponent extends NestedComponent {
     this.redraw();
   }
 
-  saveRow(rowIndex) {
+  saveRow(rowIndex, dialog) {
```

Comment:
Instead of this, I would prefer to contain the dialog.close within the ```addRowModal``` method. Basically turn it into this.

```js
addRowModal(rowIndex) {
    const formComponents =  this.ce('div');
    formComponents.innerHTML = this.renderComponents(this.editRows[rowIndex].components);
    const dialog = this.component.modal ? this.createModal(formComponents) : undefined;
    dialog.refs.dialogContents.appendChild( this.ce('button', {
      class: 'btn btn-primary',
      onClick: () => {
        this.saveRow(rowIndex);
        dialog.close();
      },
    }, this.component.saveRow || 'Save'));
    this.attachComponents(dialog, this.editRows[rowIndex].components);
  }	  }
```